base-system/usrroot/usr/lib/hsync/libhapply.so

```diff
@@ -404,3 +405,22 @@ set_wallpaper() {
 	log "-Couldn't find any background to restore"
 	return 1 # error
 }
+
+set_scheduled_shutdown() {
+	SHUTDOWN_TIME="$(get_directives_var Global ShutdownTime)"
+	log "Cancelling scheduled shutdown (if any)"
+	shutdown -c
+	if [ "$SHUTDOWN_TIME" = "none" ]; then
+		return 0
+	fi
+	SHUTDOWN_TIME_EPOCH=$(date -d "$SHUTDOWN_TIME" +%s)
+	CURRENT_TIME_EPOCH=$(date +%s)
+	SECONDS_TO_SHUTDOWN=$((SHUTDOWN_TIME_EPOCH-CURRENT_TIME_EPOCH))
+	if [ "$SECONDS_TO_SHUTDOWN" -lt 0 ]; then
+		log "Current shutdown is for a time that already happened, skiping scheduled shutdown."
+		exit 0
+	fi
+	MINUTES_TO_SHUTDOWN=$((SECONDS_TO_SHUTDOWN/60))
+	log "Scheduling shutdown in $MINUTES_TO_SHUTDOWN minute(s)"
+	shutdown +"$MINUTES_TO_SHUTDOWN"
```

Comment:
Why not to schedule this with the `at` command?

Reply:
If we end using this maybe the only problem could be erasing scheduled applies.

Reply:
> Why not to schedule this with the `at` command?

I think is less prone to unexpected shutdowns because based on [this](https://unix.stackexchange.com/a/465323) someone says

> if you use at, the job will be stored and if the system is shutdown by some other means before 18:00, upon booting again the job will still be waiting to be run; if the time is already passed, the shutdown will be performed immediately which could be quite unexpected.

Reply:
> If we end using this maybe the only problem could be erasing scheduled applies.

You mean the at, or the shutdown command? :thinking:

Reply:
> > Why not to schedule this with the `at` command?
> 
> I think is less prone to unexpected shutdowns because based on [this](https://unix.stackexchange.com/a/465323) someone says
> 
> > if you use at, the job will be stored and if the system is shutdown by some other means before 18:00, upon booting again the job will still be waiting to be run; if the time is already passed, the shutdown will be performed immediately which could be quite unexpected.

Interesting, then what happens if the system changes the clock. Let's suppose that system have 10am and shutdown is scheduled to shutdown at 12pm, then the system clock changes to 11am. The system will shutdown at 12pm or at 1pm?

Reply:
After some testings, seems like in the example you mention, the system will still shutdown at 12pm as scheduled without being affected by the clock changing.